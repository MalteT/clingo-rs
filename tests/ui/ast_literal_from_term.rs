use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let mut term = ast::Term::from(sym);
    let lit = ast::Literal::from_term(ast::Sign::None, &term);
    term = ast::Term::from(sym);
    let _end = (lit,term);
}