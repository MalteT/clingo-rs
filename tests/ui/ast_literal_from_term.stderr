error[E0506]: cannot assign to `term` because it is borrowed
 --> $DIR/ast_literal_from_term.rs:7:5
  |
6 |     let lit = ast::Literal::from_term(ast::Sign::None, &term);
  |                                                        ----- borrow of `term` occurs here
7 |     term = ast::Term::from(sym);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `term` occurs here
8 |     let _end = (lit,term);
  |                 --- borrow later used here

For more information about this error, try `rustc --explain E0506`.
